ynput/OpenPype
# Substance Painter: Allow users to set texture resolutions when loading mesh to create project

## Changelog Description
This PR is to add the support of template settings in the mesh loaders for Substance project creation. User can customize and add template settings in AYON settings and apply it through the option mode(the button with memo icon).

## Additional info
This PR also refactors the original loader's setting without directly import qargparse.
Prefers user to use options (memo icon) to set up their preferences as the prompt dialog might not set up the project as expected(It is still experimental).

## Testing notes:
1. Customize the template with a template name in Substance Painter OP setting
2. Launch Substance Painter
3. Click Load..
4. It would pop up the dialog to allow you to choose your template settings
5. Click Ok
![image](https://github.com/ynput/ayon-core/assets/64118225/957929f5-a08d-4f3d-bf31-5f4a87e73eed)

6. It should apply all the settings you set in AYON settings
